### ISSUE TITLE
Uptime percentage

### ISSUE THREAD

**Issue by camilonova** [TwinProduction/gatus#62]
There is any way is calculated automatically so we can get it?

**Comment by TwinProduction** [comment]
When you say "get it", do you mean as in have the percentage exposed through an API for other applications to consume, or do you mean as in have it displayed on the dashboard?

I've been thinking of adding something like a "more information" modal that could display similar information at some point.

**Comment by camilonova** [comment]
Having the percentage exposed through the API

**Comment by TwinProduction** [comment]
I can see this being useful. 
Would it be per service, overall, or both?

**Comment by TwinProduction** [comment]
FYI, I got a bit inspired yesterday and decided to start working on this.

I also decided to implement dynamic uptime badges.

The first version will likely come out today, but the uptime is not displayed on the dashboard yet, so it will only be available through the API and through the badges.

**Comment by TwinProduction** [comment]
Sneak peak: 
![Uptime 1h](https://status.twinnation.org/api/v1/badges/uptime/1h/group-core-service-twinnation%20-%20external.svg)
![Uptime 24h](https://status.twinnation.org/api/v1/badges/uptime/24h/group-core-service-twinnation%20-%20external.svg)
![Uptime 7d](https://status.twinnation.org/api/v1/badges/uptime/7d/group-core-service-twinnation%20-%20external.svg)

I just pushed the first version I talked about earlier in `master`, so this is currently only available in the `twinproduction/gatus:latest`.

As for your original question, the uptime is now exposed through the API (`/api/v1/statuses`):
```json
{
  "core_twinnation - article 43": {
    "name": "twinnation - article 43",
    "group": "core",
    "results": ["omitted for brevity"],
    "uptime": {
      "7d": 1,
      "24h": 1,
      "1h": 1
    }
}
```

@camilonova While it's not displayed on the frontend yet, does this satisfy your initial request (to expose the uptime through the API)?

**Comment by progressify** [comment]
> Having the percentage exposed through the API. Something like:
> ![image](https://user-images.githubusercontent.com/5376494/102575607-533b2300-40a8-11eb-9e91-63848ae803a1.png)

statping like xD

**Comment by TwinProduction** [comment]
Released part of this feature in [v1.11.0](https://github.com/TwinProduction/gatus/releases/tag/v1.11.0)

For now, the API returns the uptime percentage of each services, which was the initial request. As I've also previously mentioned, I decided to add support for uptime badges.

In the future, I will also probably add the average response on the UI, but I haven't gotten around to it yet.

**Comment by camilonova** [comment]
Thank you
